hathor/utils/dict.py

```diff
@@ -0,0 +1,29 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+def deep_merge(first: dict, second: dict) -> None:
```

Comment:
Why in place? I feel that returning a new dict might be more useful. I don't a have strong opinion about it, though.

Reply:
I agree it's more useful, I just didn't need it. Changed in https://github.com/HathorNetwork/hathor-core/pull/618/commits/32b1a3444a4c4c916f89713fed7fe8886f358e69

Reply:
Great! Thanks!